coreneuron/io/lfp.hpp

```diff
@@ -0,0 +1,226 @@
+#ifndef LFP_H
+#define LFP_H
+
+#include <cmath>
+#include <array>
+#include <iostream>
+#include <vector>
+
+#include <boost/math/constants/constants.hpp>
+
+#include "coreneuron/utils/nrn_assert.h"
+#include "coreneuron/mpi/nrnmpidec.h"
+
+namespace coreneuron {
+
+template <typename T>
+using Array3 = std::array<T, 3>;
+
+template <typename Point3D, typename F>
+F dot(const Point3D& p1, const Point3D& p2) {
+    return p1[0] * p2[0] + p1[1] * p2[1] + p1[2] * p2[2];
+}
+
+template <typename Point3D, typename F>
+F norm(const Point3D& p1) {
+    return std::sqrt(dot<Point3D, F>(p1, p1));
+}
+
+template <typename Point3D, typename F>
+Array3<F> barycenter(const Point3D& p1, const Point3D& p2) {
+    return {0.5 * (p1[0] + p2[0]), 0.5 * (p1[1] + p2[1]), 0.5 * (p1[2] + p2[2])};
+}
+
+template <typename Point3D, typename F>
+Array3<F> axp(const Point3D& p1, const F alpha, const Point3D& p2) {
```

Comment:
What about Point3D being a naked pointer?

Reply:
Ok. I designed the class at the time without knowing the type of classes I would encounter. It would make sense now to remove the template.

Reply:
in addition, this function could be split into:

- multiplication by scalar
- point addition

if we implement in the point class these 2 operators we could write the operations in a much nicer way so we do not need to remember, for example, that this function is p1 + a*p2 instead of a*p1 + p2

Reply:
I think the idea of having one function combined is for speed, possibly premature optimization, but is it a very common operation and to me it makes sense to combine it. The name of the function tells you the order of the operations `paxpy` p(type) a(alpha) x(first input) p(plus) y(second input), basically follows blas naming convention.

Reply:
I followed @castigli advice here.